daceml/pytorch/module_codegen.py

```diff
@@ -88,35 +96,64 @@ def initialize_outputs_code(module: 'daceml.pytorch.DaceModule',
     return code
 
 
-def argument_codegen(module: 'daceml.pytorch.DaceModule',
-                     input_names: List[str],
-                     output_names: List[str]) -> Tuple[str, str, str]:
+def argument_codegen(
+        sdfg: dace.SDFG,
+        clean_weights: Dict[str, torch.Tensor],
+        input_names: List[str],
+        output_names: List[str],
+        guard_contiguous: Optional[List[str]] = None) -> Tuple[str, str, str]:
     """ Generate the code that grabs the pointers of inputs and outputs.
 
+        The names of the tensors will match the SDFG tensor names. Tensors that are not created by us (i.e. inputs)
+        should be named {sdfg_name}_ first, and then .contiguous() will be called on them to yield the tensor that we
+        require. This is the case for all tensors in ``guard_contiguous``.
+
         :param module: the module
+        :param clean_weights: the constant weights of the SDFG.
+        :param input_names: names of inputs to the torch function.
+        :param output_names: names of outputs to the torch function.
+        :param guard_contiguous: a subset of input_names to call .contiguous on. If None, all input names will be
+                                 guarded.
         :return: the code for initializing the argument, the sdfg arguments in order, and the init call arguments
     """
-    arglist = module.sdfg.arglist()
+    arglist = sdfg.arglist()
+
+    guard_contiguous = set(guard_contiguous or input_names)
 
     # initialize the inputs and outputs
-    ptr_init_code = "\n    // setup input and output pointers\n    "
-    # inputs: make these contiguous if they're not
-    ptr_init_code += '\n    '.join(
-        f"{arglist[name].dtype.ctype} *{name}_ptr = {name}_.contiguous().data_ptr<{arglist[name].dtype.ctype}>();"
-        for name in input_names)
-    # outputs
-    ptr_init_code += '\n    '.join(
-        f"{arglist[name].dtype.ctype} *{name}_ptr = {name}_.data_ptr<{arglist[name].dtype.ctype}>();"
+    ptr_init_code = "\n// setup input and output pointers\n"
+    for name in input_names:
+        ctype = arglist[name].dtype.ctype
+        if isinstance(arglist[name], data.Array) or dt.can_access(
+                dt.ScheduleType.GPU_Device, arglist[name].storage):
+            if name in guard_contiguous:
+                ptr_init_code += '\n' + f"Tensor {name} = {name}_.contiguous();"
+            ptr_init_code += '\n' + f"{ctype} *{name}_ptr = {name}.data_ptr<{ctype}>();"
+
+        elif isinstance(arglist[name], data.Scalar):
+            if name in guard_contiguous:
+                ptr_init_code += '\n' + f"{ctype} {name}_ptr = {name}_.item().to<{ctype}>();"
```

Comment:
same here, but the underscore in the definition above was removed. is this on purpose?

Reply:
See above